metrics/influx/influx_test.go

```diff
@@ -31,6 +30,33 @@ func TestCounter(t *testing.T) {
 	}
 }
 
+func ExampleCounter() {
+	in := New(map[string]string{"a": "b"}, influxdb.BatchPointsConfig{}, log.NewNopLogger())
+	counter := in.NewCounter("influx_counter")
+	counter.Add(10)
+	counter.With("error", "true").Add(1)
+	counter.With("error", "false").Add(2)
+	counter.Add(50)
+
+	client := &bufWriter{}
+	in.WriteTo(client)
+
+	expectedLines := []string{
+		`(influx_counter,a=b count=60) [0-9]{19}`,
+		`(influx_counter,a=b,error=true count=1) [0-9]{19}`,
+		`(influx_counter,a=b,error=false count=2) [0-9]{19}`,
+	}
+
+	if err := extractAndPrintMessage(expectedLines, client.buf.String()); err != nil {
+		fmt.Println(err.Error())
+	}
+
+	// Output:
+	// influx_counter,a=b count=60
+	// influx_counter,a=b,error=true count=1
+	// influx_counter,a=b,error=false count=2
+}
```

Comment:
It doesn't fail because it extracts the timestamp before printing each line. so, instead of printing `influx_counter,a=b count=60 1234567891234567890` (sample timestamp), it just prints `influx_counter,a=b count=60`.

If any of the given patterns doesn't match, it also prints an error message and the example fails